src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,40 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else if self.is_initialized() {
```

Comment:
The problem is that, for uninitialized peers:

1. A ready event notifies that a leader has emerged, then it "has leader".
2. This peer get isolated, and no ready event will be triggered. This peer still keeps its state as "has leader". 

Then we don't know when this peer could be gc, even do checking it in raft tick.